### PR TITLE
Sqlite date functions

### PR DESCRIPTION
**Pull request by tarjei**
Hi, I've missed Date(), Year()  and Month() functions that work across Mysql and Sqlite for quite some time. Since you've done all the hard work, I thought I'd just build on top of it. Here are implementations for Sqlite of seven normal date functions.

### CODE REVIEW

**Comment by stevelacey** [comment]
Awesome! Can you clear up the superfluous whitespace and commented out code and I'll take a proper look to get this merged in

**Comment by tarjei** [comment]
If there are whitespace cleanups I have missed (or not thought about) please point them out. I would like to keep the comments in sqlite.yml as a todolist for further expansions :)

**Comment by stevelacey** [comment]
Also check out the issues Travis found

File: src/Query/Sqlite/Date.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace DoctrineExtensions\Query\Sqlite;
+
+/**
+ * @author Steve Lacey <steve@stevelacey.net>
+ */
+class Date extends AbstractStrfTime
+{
+
+    protected function getFormat()
```
Comment (stevelacey) [inline src/Query/Sqlite/Date.php]: No need for double new line before this

File: tests/Query/Sqlite/StrfTimeTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * User: tarjei
+ * Date: 23.06.15 / 12:23
+ */
+
+namespace DoctrineExtensions\Tests\Query\Sqlite;
+
+
+use DoctrineExtensions\Tests\Query\SqliteTestCase;
```
Comment (stevelacey) [inline tests/Query/Sqlite/StrfTimeTest.php]: Or here

File: src/Query/Sqlite/AbstractStrfTime.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        $parser->match(Lexer::T_IDENTIFIER);
+        $parser->match(Lexer::T_OPEN_PARENTHESIS);
+
+        $this->date = $parser->ArithmeticPrimary();
+
+        $parser->match(Lexer::T_CLOSE_PARENTHESIS);
+    }
+
+    abstract protected function getFormat();
+}
```
Comment (stevelacey) [inline src/Query/Sqlite/AbstractStrfTime.php]: Files should end with a new line

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 /composer.lock
 /tests/phpunit.xml
 /vendor
+.idea
```
Comment (stevelacey) [inline .gitignore]: Only project specifics should go in this gitignore, you can ignore stuff related to your personal setup with a global gitignore: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: src/Query/Sqlite/Month.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace DoctrineExtensions\Query\Sqlite;
+
+/**
+ * @author Rafael Kassner <kassner@gmail.com>
+ * @author Sarjono Mukti Aji <me@simukti.net>
+ */
```
Comment (stevelacey) [inline src/Query/Sqlite/Month.php]: Trash all author refs or add your own

**Comment by stevelacey** [comment]
Looking good :+1: a few random points and obviously fix the issue Travis found

**Comment by tarjei** [comment]
Hmm, I got a problem with the errors that travis is making that go like this:
```
-'SELECT strftime('%Y', d0_.created) AS sclr_0 FROM Date d0_'
+'SELECT strftime('%Y', d0_.created) AS sclr0 FROM Date d0_'
```

The tests work on my local computer, but there the other (MySQL) tests fail with the same error. Is there a way to ensure the naming is consistent? I.e. either sclr_0 or sclr0?

**Comment by stevelacey** [comment]
I expect it is something that changed between Doctrine versions – I think
in other places we've used regex to allow for the difference, or checked
the version and adjusted the check.

On 24 June 2015 at 19:52, Tarjei Huse <notifications@github.com> wrote:

> Hmm, I got a problem with the errors that travis is making that go like
> this:
>
> -'SELECT strftime('%Y', d0_.created) AS sclr_0 FROM Date d0_'
> +'SELECT strftime('%Y', d0_.created) AS sclr0 FROM Date d0_'
>
> The tests work on my local computer, but there the other (MySQL) tests
> fail with the same error. Is there a way to ensure the naming is
> consistent? I.e. either sclr_0 or sclr0?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beberlei/DoctrineExtensions/pull/152#issuecomment-114979143>
> .
>



-- 

Steve Laceyhttp://stevelacey.net@stevelacey
<http://www.twitter.com/stevelacey> / steve@stevelacey.net

**Comment by tarjei** [comment]
OK, is there anything else needed on this PR now? 

I just added Doctrine 2.5 to the .travis.yml file.

**Comment by stevelacey** [comment]
Unfortunately, the library is yet to support Doctrine 2.5, and this branch now conflicts with master. Can you remove the change to .travis.yml and resolve the conflict? We're working on 2.5 support in #160.

**Comment by stevelacey** [comment]
I just merged support for testing against Doctrine 2.5 into master, including fixes for the MySQL test column alias expectations – but it looks like you've addressed that here anyway.

Either way, if you merge in master I expect this is good to go.

**Comment by tarjei** [comment]
OK, I rebased and removed the .travis addition.

**Comment by stevelacey** [comment]
@tarjei I am in two minds now about this, I kind of like your column alias fix for older Doctrine, but I already bodged it fixed for older versions in https://github.com/beberlei/DoctrineExtensions/blob/master/tests/Query/MysqlTestCase.php#L57-L65 – and I quite like the simplicity of that fix, it allows us to write tests to match Doctrine 2.5+ (like you did) and worry about hacking older ones fixed elsewhere.

What are your thoughts? I really want to get your Sqlite stuff in, but I think I want to put more thought into the column aliasing if we're going to switch away from my lightweight fix – and I quite like the lack of verbosity in my implementation.

Any chance you can separate out these two pieces of work and apply a similar fix to mine to your base SqliteTestCase class?

**Comment by tarjei** [comment]
Hmm, looking at your fix, I think it has a nice simplicity to it. I'll see
if I can do another rebase in that direction :)

2015-08-06 10:46 GMT+02:00 Steve Lacey <notifications@github.com>:

> @tarjei <https://github.com/tarjei> I am in two minds now about this, I
> kind of like your column alias fix for older Doctrine, but I already bodged
> it fixed for older versions in
> https://github.com/beberlei/DoctrineExtensions/blob/master/tests/Query/MysqlTestCase.php#L57-L65
> – and I quite like the simplicity of that fix, it allows us to write tests
> to match Doctrine 2.5+ (like you did) and worry about hacking older ones
> fixed elsewhere.
>
> What are your thoughts? I really want to get your Sqlite stuff in, but I
> think I want to put more thought into the column aliasing if we're going to
> switch away from my lightweight fix – and I quite like the lack of
> verbosity in my implementation.
>
> Any chance you can separate out these two pieces of work and apply a
> similar fix to mine to your base SqliteTestCase class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beberlei/DoctrineExtensions/pull/152#issuecomment-128293923>
> .
>



-- 
__________________
Tarjei Huse
Mobil: 920 63 413

**Comment by tarjei** [comment]
The failing tracvis check is not related to the PR, but a test that is a bit flakey since it relies on the systems time.

**Comment by stevelacey** [comment]
@tarjei yep, aware of that problem – I'll poke your build

**Comment by stevelacey** [comment]
@tarjei okay, this looks good, you might want to consider updating [tests/Config/MysqlYamlTest.php](https://github.com/beberlei/DoctrineExtensions/blob/cce647a/tests/Config/MysqlYamlTest.php) to also check for your Sqlite functions in the docs – I am happy to merge this if you want to do it separately however.

**Comment by tarjei** [comment]
Please do :P

I do not have time to add the test today, but I would love the feeling of
finally getting this into master.

2015-08-06 13:47 GMT+02:00 Steve Lacey <notifications@github.com>:

> @tarjei <https://github.com/tarjei> okay, this looks good, you might want
> to consider updating tests/Config/MysqlYamlTest.php
> <https://github.com/beberlei/DoctrineExtensions/blob/cce647a/tests/Config/MysqlYamlTest.php>
> to also check for your Sqlite functions in the docs – I am happy to merge
> this if you want to do it separately however.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beberlei/DoctrineExtensions/pull/152#issuecomment-128340032>
> .
>



-- 
__________________
Tarjei Huse
Mobil: 920 63 413

**Comment by stevelacey** [comment]
@tarjei I will trust you to submit a pull request adding it in later :wink: thanks for your contribution!

**Pull request merged**